### PR TITLE
[Serve] Add metric to track ongoing requests in proxies

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change adds a metric (`"serve_num_ongoing_http_requests"`) to track the number of ongoing requests in each HTTP Proxy.

Link to updated documentation: https://anyscale-ray--38334.com.readthedocs.build/en/38334/serve/monitoring.html#built-in-ray-serve-metrics

This is a signed version of #38283.

## Related issue number

N/A

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - This test updates `test_queued_queries_disconnected` to also test the new metric.

### CODE REVIEW

File: python/ray/serve/_private/http_proxy.py [hunk 1]
```diff
@@ -295,6 +298,19 @@ def get_handle(name):
...
             ),
         )
+
+        self.num_ongoing_requests_gauge = metrics.Gauge(
+            name="serve_num_ongoing_requests",
+            description="The number of ongoing requests in this HTTP Proxy.",
+            tag_keys=("actor_name", "node_id", "node_ip_address"),
+        ).set_default_tags(
+            {
+                "actor_name": actor_name,
```
Comment (edoakes) [inline python/ray/serve/_private/http_proxy.py:308]: I think actor name is totally redundant with `node_id`

**Review by edoakes — COMMENTED** [review]
Looks good, just suggest to remove the actor name tag

**Review by GeneDer — APPROVED** [review]
:shipit:

File: python/ray/serve/_private/http_proxy.py (hunk 1 above)
Comment (shrekris-anyscale) [reply python/ray/serve/_private/http_proxy.py]: Good call, I removed it.

**Comment by shrekris-anyscale** [comment]
I added the metric to the documentation.

**Review by edoakes — APPROVED** [review]

**Comment by shrekris-anyscale** [comment]
The test failures are unrelated:

* `Windows Build & Test` ([`test_actor_bounded_threads`](https://buildkite.com/ray-project/oss-ci-build-pr/builds/31596#0189e5de-5334-4c80-b44b-f9cde78196a9/6-12140)) is flakey on `master`:

Screen Shot 2023-08-11 at 4 08 39 PM

**Comment by shrekris-anyscale** [comment]
@edoakes This change is ready to merge.

File: doc/source/serve/monitoring.md [hunk 2]
```diff
@@ -317,6 +317,10 @@ The following metrics are exposed by Ray Serve:
        * error_code
        * method
      - The number of non-200 HTTP responses.
+   * - ``serve_num_ongoing_requests`` [*]
```
Comment (sihanwang41) [inline doc/source/serve/monitoring.md:320]: how about changing it to `serve_num_ongoing_http_requests`? since we will have gRPC proxy very soon.

Comment (shrekris-anyscale) [reply doc/source/serve/monitoring.md]: Good idea, I renamed it.

**Pull request merged**